### ISSUE TITLE
Is it intended? Column loaded twice if listed twice in "columns=" when reading a Parquet file.

### ISSUE THREAD

**Issue by bsesar** [dask/dask#9056]
**What happened**: When loading a Parquet file, I specified a column twice in the "columns=" argument and the column was loaded twice, that is, there were two columns in the resulting DataFrame with the same name and data.

**What you expected to happen**: Even though I specified the column twice, I expected that the column would only be loaded once.

**Minimal Complete Verifiable Example**:

```python
# assume a Parquet file has a column called "test" and then load it twice
ddf = dd.read_parquet("file.parquet", columns=['test', 'test'])
ddf.head()
```

**Anything else we need to know?**:

**Environment**:

- Dask version: 2022.05.0
- Python version: 3.8.13
- Operating System: Windows Server 2016
- Install method (conda, pip, source): conda

**Comment by jorloplaz** [comment]
This matches with the discussion in #9122 : IMHO Dask should prevent at all costs building DataFrames with duplicate column names.

**Comment by dhirensr** [comment]
@jsignell : when I tried to reproduce I'm getting the same behaviour for pandas and dask. wondering if it's still an issue if it's consistent with pandas?
 I used the minimal example above.

```
                      id   id         x
timestamp                              
2000-01-01 00:00:00  956  956 -0.731324
2000-01-01 00:00:01  994  994  0.301500
2000-01-01 00:00:02  996  996 -0.871429
2000-01-01 00:00:03  926  926 -0.963692
2000-01-01 00:00:04  989  989  0.801868
                      id   id         x
timestamp                              
2000-01-01 00:00:00  956  956 -0.731324
2000-01-01 00:00:01  994  994  0.301500
2000-01-01 00:00:02  996  996 -0.871429
2000-01-01 00:00:03  926  926 -0.963692
2000-01-01 00:00:04  989  989  0.801868
```

**Comment by jorloplaz** [comment]
I believe that `read_parquet` (both in Pandas and Dask) should have some control of the `columns` parameter, so that either:

- Something like `sorted(set(columns))` is done silently under the hood, to remove duplicate column names before loading them.
- An error is raised whenever `len(set(columns)) < len(columns)`, asking the user explicitly to remove duplicate column names.

At a lower level anyway, both `pd.DataFrame` and `dd.DataFrame` should control that they have no duplicate column names. This means for example that you cannot call `rename` on columns and use already-existing column names.

Honestly, I can't think of any use case where it makes sense to have different columns, but named equally...

**Comment by dhirensr** [comment]
@jorloplaz  : I agree with the behaviour, but to make it consistent we'd need to change pandas as well. because in case of duplicate columns today both pandas and dask are behaving the same. 
do you think it's worth starting a change in dask? to maintain parity, we'd need to even request a` pandas change.

**Comment by jorloplaz** [comment]
Yes and no. First of all, I don't think that Pandas should allow either having duplicate column names, but they have the point that columns and rows are "symmetrical", in the sense that you can do for example `df.T`, and then, if you allow duplicates in the `index`, you should also allow duplicates in the `columns`.

However, in Dask that symmetry is lost. You can't exchange easily `index` for `columns`, because the `index` is used for internal partitioning, while `columns` are not. Thus, clearly Dask is more into the scheme "rows are observations, columns are features, period". So IMHO it makes more sense to force that `columns` have no duplicates at all, even if Pandas lets you have them.

Besides, forcing this simplifies a lot of things:

- `df[column_name]`  always returns a `dd.Series`, same as `df.iloc[:, n]`.
- `df.groupby(column_name)` always gives a `dd.SeriesGroupBy`.
- `df.compute()` guarantees that the resulting `pd.DataFrame` has no duplicate columns.
- etc, etc
